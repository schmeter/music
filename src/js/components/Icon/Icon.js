import React from 'react';
import PropTypes from 'prop-types';

class Icon extends React.Component {
    render() {
        const { id } = this.props;
        return <i className={`fa-${id} fa`} />;
    }
}

Icon.propTypes = {
    id: PropTypes.string.isRequired
};

export default Icon;
