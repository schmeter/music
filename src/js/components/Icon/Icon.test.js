import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Icon from './Icon';

describe('Icon', () => {
    it('renders correctly', () => {
        const props = {
            id: 'test',
            spin: true,
        };
        const component = shallow(<Icon {...props} />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
