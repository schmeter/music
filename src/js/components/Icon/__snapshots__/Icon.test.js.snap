// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  className="fa-test fa fa-spin"
/>
`;
